extensions/amp-accordion/1.0/accordion.js

```diff
@@ -88,19 +92,31 @@ function AccordionWithRef(
           expandSingleSection
         );
       });
-      return () => setExpandedMap((expandedMap) => omit(expandedMap, id));
+      eventMap.current[id] = {onExpand, onCollapse};
+      return () => {
+        setExpandedMap((expandedMap) => omit(expandedMap, id));
+        omit(/** @type {!Object} */ (eventMap.current), id);
+      };
     },
     [expandSingleSection]
   );
 
   const toggleExpanded = useCallback(
     (id) => {
+      const newValue = !expandedMap[id];
       setExpandedMap((expandedMap) => {
-        const newValue = !expandedMap[id];
         return setExpanded(id, newValue, expandedMap, expandSingleSection);
       });
+      const onExpand = eventMap.current[id] && eventMap.current[id].onExpand;
+      const onCollapse =
+        eventMap.current[id] && eventMap.current[id].onCollapse;
+      if (newValue && onExpand) {
+        onExpand();
+      } else if (onCollapse) {
+        onCollapse();
+      }
     },
-    [expandSingleSection]
+    [expandSingleSection, expandedMap]
```

Comment:
It's very important that the `toggleExpanded` does not directly depend on the `expandedMap`. This is because the `toggleExpanded` mutates it. So it becomes problematic that the callback that mutates some state is immediately gets re-created when this state is updated - it's a recursive dependency that we should avoid.

Reply:
Thanks, this has been updated!